pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -17,6 +21,19 @@ type clusterNameData struct {
 	mutex       sync.Mutex
 }
 
+// Provider is a generic function to grab the clustername and return it
+type Provider func() (string, error)
+
+// ProviderCatalog holds all the various kinds of clustername providers
+var ProviderCatalog = make(map[string]Provider)
+
+// RegisterClusterNameProviders registers a hostname provider as part of the catalog
+func RegisterClusterNameProviders() {
```

Comment:
nit: you can use a [package init function](https://golang.org/doc/effective_go.html#init) to populate the map, instead of having a function to call

Reply:
thanks for the best practice ;)
there is already an init function in the module I'll use it